lib/rules/no-deprecated-router-transition-methods.js

```diff
@@ -0,0 +1,201 @@
+'use strict';
+
+const emberUtils = require('../utils/ember');
+const { getImportIdentifier } = require('../utils/import');
+const Stack = require('../utils/stack');
+const types = require('../utils/types');
+const decoratorUtils = require('../utils/decorators');
+
+function getBaseFixSteps(fixer, context, currentClass) {
+  const fixSteps = [];
+  const sourceCode = context.getSourceCode();
+  let serviceInjectImportName = currentClass.serviceInjectImportName;
+  let routerServicePropertyName = currentClass.routerServicePropertyName;
+
+  if (!serviceInjectImportName) {
+    fixSteps.push(
+      fixer.insertTextBefore(
+        sourceCode.ast,
+        "import { inject as service } from '@ember/service';\n"
+      )
+    );
+
+    serviceInjectImportName = 'service';
+  }
+
+  if (!routerServicePropertyName) {
+    fixSteps.push(
+      fixer.insertTextBefore(
+        currentClass.node.body.body[0],
+        `@${serviceInjectImportName}('router') router;\n`
+      )
+    );
+
+    routerServicePropertyName = 'router';
+  }
+
+  return { fixSteps, routerServicePropertyName };
+}
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'enforce usage of router service transition methods',
+      category: 'Deprecations',
+      recommended: false,
+      url: 'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-deprecated-router-transition-methods.md',
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      main: 'Calling "{{methodUsed}}" in {{moduleType}}s is deprecated, call {{desiredMethod}} on the injected router service instead.',
+    },
+  },
+
+  create(context) {
+    // State being tracked for this file.
+    let serviceInjectImportName = undefined;
+    let routerServicePropertyName = undefined;
+    let isValidModule = false;
+
+    // State being tracked for the current class we're inside.
+    const classStack = new Stack();
+
+    // This rule does not apply to test files or non class modules
+    if (emberUtils.isTestFile(context.getFilename())) {
```

Comment:
According to test coverage checks, this line is not tested. Can you test it? And also be sure to monitor the test coverage to ensure every line in the new rule is covered.

```
yarn test --coverage lib/rules/no-deprecated-router-transition-methods.js
```

Reply:
After looking through, I'm going to remove this test check.

There are cases where you may have module definitions in tests and we don't want to ignore them.

Though it's unlikely for teams to define mock controllers/routes, it is still possible.